wavedump/src/fit-histograms

```diff
@@ -0,0 +1,566 @@
+                h.GetListOfFunctions().Clear()
+                spec = ROOT.TSpectrum()
+                if(args.BV==42):
+                    n_pks = spec.Search(h, 1, "", 0.01)
+                    #n_pks = spec.Search(h, 1, "", 0.999)
+                elif(args.BV>42):
+                    n_pks = spec.Search(h, 1, "", 0.005)
+                else:
+                    # print("Wrong BV<42")
+                    sys.exit("Wrong BV<42")
+
+
+                x_pos = spec.GetPositionX()
+                y_pos = spec.GetPositionY()
+
+                x_pos = np.array([x_pos[i] for i in range(n_pks)])
+                y_pos = np.array([y_pos[i] for i in range(n_pks)])
+                print("Peak charge full list:")
+                print(x_pos)
+                ind = np.argsort(x_pos)
+                x_pos = x_pos[ind]
+                y_pos = y_pos[ind]
+                                      
+                n = len(x_pos)
+                
+                # peak = x_pos[-1]
+               
+                f1 = None
+                win = 0.3 * h.GetStdDev()
+                for i in range(n-1, -1, -1):
+                    peak = x_pos[i]
+                    if peak < x_pos[0] + 0.01*h.GetStdDev():
+                        continue
+                    print(f'Fitting this peak! {peak}')
+                    f1 = ROOT.TF1("f1","gaus", peak-win, peak+win)
+                    r = h.Fit(f1, 'ILMSR+')
+                    r = r.Get()
+                    if not r.IsValid():
+                        continue
+                    if f1.GetParameter(0) > h.GetEntries() / 200:
+                        break
+                if f1 == None:
+                    peak = x_pos[-1]
+                    f1 = ROOT.TF1("f1","gaus", peak-win, peak+win)
+                    h.Fit(f1, 'ILMSR+')
+
+
+                #*** Store values in txt/csv file
+                print("511 peak = %.4f +/- %.4f" % (f1.GetParameter(1), f1.GetParError(1)))
+                save_as_csv('sodium', f1.GetParameter(1), f1.GetParError(1), h.GetName())
+
+                h.Write()
+                if args.pdf:
+                    print("*** Print PDF 1:")
+                    canvas.append(ROOT.TCanvas("c%i" % canvas_count))
+                    canvas_count += 1
+                    h.Draw()
+                    canvas[-1].Update()
+                    prname=filename[:len(filename)-5]
+                    canvas[-1].Print("./FiguresFit/"+prname+"_Key_"+str(keycount)+".pdf")
+                elif args.plot:
+                    canvas.append(ROOT.TCanvas("c%i" % canvas_count))
+                    canvas_count += 1
+                    h.Draw()
+                    canvas[-1].Update()
```

Comment:
Can you specify both plot and pdf? If so, these shouldn't be in an elif. Also, see comments previously in analyze-histograms.